src/parser/jobs/ast/modules/Synastry.tsx

```diff
@@ -0,0 +1,77 @@
+import {Plural, Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import {CastEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Combatants from 'parser/core/modules/Combatants'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SEVERITIES = {
+	WASTED_ST_HEAL_TIERS: {
+		1: SEVERITY.MINOR,
+		10: SEVERITY.MEDIUM,
+	},
+}
+
+const GCD_ST_HEAL = [
+	ACTIONS.BENEFIC.id,
+	ACTIONS.BENEFIC_II.id,
+]
+
+// Ripped off from WHM and converted to TSX
+export default class Synastry extends Module {
+	static handle = 'synastry'
+
+	@dependency private combatants!: Combatants
+	@dependency private suggestions!: Suggestions
+
+	private lastUse = 0
+	private uses = 0
+	private totalHeld = 0
+	private nonSynastryHeals = 0
+
+	protected init() {
+		this.addHook('cast', {abilityId: ACTIONS.SYNASTRY.id, by: 'player'}, this.onCast)
+		this.addHook('cast', {abilityId: GCD_ST_HEAL, by: 'player'}, this.onSingleTargetHealCast)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onSingleTargetHealCast(event: CastEvent) {
+		// Ignore if Synastry is still on CD or we already have it up
+		if (this.combatants.selected.hasStatus(STATUSES.SYNASTRY_SELF.id)
+		|| this.lastUse + (ACTIONS.SYNASTRY.cooldown * 1000) > event.timestamp) {

```

Comment:
Not a huge issue since you're using `lastUse` elsewhere to track the hold time, but it's worth noting that the core Cooldowns module tracks this for you and you can simplify the manual tracking into `this.cooldowns.getCooldownRemaining(ACTIONS.SYNASTRY.id) > 0`.